app/Services/PirepService.php

```diff
@@ -489,6 +489,8 @@ public function submit(Pirep $pirep)
         // only update the pilot last state if they are accepted
         if ($default_state === PirepState::ACCEPTED) {
             $pirep = $this->accept($pirep);
+        } elseif ($default_state === PirepState::REJECTED) {
+            event(new PirepRejected($pirep));
```

Comment:
Just update your branch and then replace this line with the `reject()` method as below please.
```php
$pirep = $this->reject($pirep);
```
All will be fine then 😸

Reply:
@avs-code

Reply:
Before do push, is this right?
```php
        // only update the pilot last state if they are accepted
        if ($default_state === PirepState::ACCEPTED) {
            $pirep = $this->accept($pirep);
        } elseif ($default_state === PirepState::REJECTED) {
            $pirep = $this->reject($pirep);
        } else {
            $pirep->state = $default_state;
        }
```